### PR TITLE
Feature/redux

### PR DESCRIPTION
**Pull request by AndreiBike**

### CODE REVIEW

File: src/client/components/header/search/Search.js [hunk 1]
```diff
@@ -39,13 +52,18 @@ class Search extends React.Component {
     let gengreClass = cn({
       'gengre-button': true,
       'active': this.state.searchByGengre.status,
-    })
+    });
+
+    let searchClass = cn('search-input-button');
```
Comment (NurbekAman) [inline src/client/components/header/search/Search.js]: move it into className without `cn`

File: src/client/components/shared/preloader/Preloader.js [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import React from 'react';
+import './Preloader.module.css';
 
 let Preloader = (props) => {
```
Comment (NurbekAman) [inline src/client/components/shared/preloader/Preloader.js:4]: const

File: src/client/components/shared/preloader/Preloader.module.css [hunk 3]
```diff
@@ -0,0 +1,10 @@
+.preloader-style {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: 1200px;
+}
+
+.preloader-style img {
+  margin: 50px 50px 50px 50px;
```
Comment (NurbekAman) [inline src/client/components/shared/preloader/Preloader.module.css:9]: margin: 50px;

File: src/client/components/sortbar/Sortbar.js [hunk 4]
```diff
@@ -9,35 +9,34 @@ const Sortbar = (props) => {
...
 
-  const clickButton = (prevState, isRatingStatus) => {
-    const { sortMode, sortByRating, sortByReleaseDate } = prevState;
-    setState({
-      sortMode,
-      sortByRating: { ...sortByRating, status: isRatingStatus },
-      sortByReleaseDate: { ...sortByReleaseDate, status: !isRatingStatus }
-    });
+  const clickButton = (isRatingStatus) => {
+    props.uploadMovies('sorting',
```
Comment (NurbekAman) [inline src/client/components/sortbar/Sortbar.js:24]: `props.uploadMovies({ sorting, isRatingStatus, searchText: props.searchText, searchBy: props.searchBy, offset: 0, total: props.total })`

File: src/client/components/sortbar/Sortbar.js [hunk 5]
```diff
@@ -9,35 +9,34 @@ const Sortbar = (props) => {
...
+      isRatingStatus ? 'vote_average' : 'release_date',
+      props.searchText,
+      props.searchBy,
+      0,
+      props.total)
   }
 
   let releaseClass = cn({
```
Comment (NurbekAman) [inline src/client/components/sortbar/Sortbar.js:32]: assign directly to toggleClassName

File: src/client/components/sortbar/Sortbar.js [hunk 6]
```diff
@@ -9,35 +9,34 @@ const Sortbar = (props) => {
...
     "release-button": true,
-    "active": state.sortByReleaseDate.status,
+    "active": (props.sortBy === 'release_date'),
   });
 
   let ratingClass = cn({
```
Comment (NurbekAman) [inline src/client/components/sortbar/Sortbar.js:37]: the same as above

File: src/client/reduxStore/actions.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
+import * as types from './actionTypes';
+
+//Uploading movies
+export function uploadMoviesAction(effect,sortBy, searchText, searchBy,offset) {
+  return {
+    type: types.UPLOAD_MOVIES,
+    effect,
```
Comment (NurbekAman) [inline src/client/reduxStore/actions.js:7]: payload: { effect, ... }

File: src/client/reduxStore/actions.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    sortBy,
+    searchText, 
+    searchBy,
+    offset,
+  }
+}
+
+export function uploadMoviesSuccsessAction(reseivedMovies,total,effect,sortBy, searchText,searchBy,offset) {
+  return {
+    type: types.UPLOAD_MOVIES_SUCCSESS,
```
Comment (NurbekAman) [inline src/client/reduxStore/actions.js:17]: the same

File: src/client/reduxStore/moviesReducer.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
+import * as types from './actionTypes';
+
+let initialState = {
```
Comment (NurbekAman) [inline src/client/reduxStore/moviesReducer.js:3]: const

File: src/client/reduxStore/moviesReducer.js [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+  searchText: '',
+  searchBy: 'title',
+  sortBy: 'release_date',
+  offset: 0,
+  total: 0,
+  isLoading: false,
+  movies: []
+}
+
+export function moviesReducer(state = initialState, action) {
```
Comment (NurbekAman) [inline src/client/reduxStore/moviesReducer.js:13]: please refactor this in order that action have only two props: type and payload

File: src/client/reduxStore/moviesReducer.js [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+      }
+
+    case types.UPLOAD_MOVIES:
+      return {
+        ...state,
+        isLoading: !state.movies.length,
+      }
+
+    case types.UPLOAD_MOVIES_SUCCSESS:
+      let movies = [];
```
Comment (NurbekAman) [inline src/client/reduxStore/moviesReducer.js:27]: use here map or reduce

File: src/client/reduxStore/moviesReducer.js [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+        searchText: action.searchText,
+        searchBy: action.searchBy,
+        sortBy: action.sortBy,
+        offset: (action.effect === 'loading') ? state.offset += 12 : 12,
+        total: action.total,
+        isLoading: false,
+        movies: (action.effect === 'loading') ? state.movies.concat(movies) : movies,
+      }
+
+    case types.UPLOAD_MOVIES_FAILED:
```
Comment (NurbekAman) [inline src/client/reduxStore/moviesReducer.js:50]: you can remove it

File: src/client/components/header/search/Search.js (hunk 1 above)
Comment (AndreiBike) [reply src/client/components/header/search/Search.js]: Done)

File: src/client/components/shared/preloader/Preloader.js (hunk 2 above)
Comment (AndreiBike) [reply src/client/components/shared/preloader/Preloader.js:4]: Done)

File: src/client/components/shared/preloader/Preloader.module.css (hunk 3 above)
Comment (AndreiBike) [reply src/client/components/shared/preloader/Preloader.module.css:9]: Corrected)

File: src/client/components/sortbar/Sortbar.js (hunk 4 above)
Comment (AndreiBike) [reply src/client/components/sortbar/Sortbar.js:24]: Corrected

File: src/client/components/sortbar/Sortbar.js (hunk 5 above)
Comment (AndreiBike) [reply src/client/components/sortbar/Sortbar.js:32]: Corrected

File: src/client/components/sortbar/Sortbar.js (hunk 6 above)
Comment (AndreiBike) [reply src/client/components/sortbar/Sortbar.js:37]: Corrected

File: src/client/reduxStore/actions.js (hunk 7 above)
Comment (AndreiBike) [reply src/client/reduxStore/actions.js:7]: Created the object

File: src/client/reduxStore/actions.js (hunk 8 above)
Comment (AndreiBike) [reply src/client/reduxStore/actions.js:17]: Done

File: src/client/reduxStore/moviesReducer.js (hunk 9 above)
Comment (AndreiBike) [reply src/client/reduxStore/moviesReducer.js:3]: Corrected

File: src/client/reduxStore/moviesReducer.js (hunk 10 above)
Comment (AndreiBike) [reply src/client/reduxStore/moviesReducer.js:13]: Refctored

**Pull request merged**